rxandroid/src/main/java/rx/android/schedulers/HandlerScheduler.java

```diff
@@ -44,45 +45,49 @@ public Worker createWorker() {
     }
 
     static class HandlerWorker extends Worker {
-
-        final Handler handler;
-
-        private final CompositeSubscription compositeSubscription = new CompositeSubscription();
+        private final Handler handler;
+        /* Writes are guarded by 'this'. */
+        private volatile boolean unsubscribed;
 
         HandlerWorker(Handler handler) {
```

Comment:
Please correct me if I'm wrong but the idea in #275 was to create `Handler` per `Worker` to perform mass cancellation of the tasks. At the moment, you share one `Handler` between all `HandlerWorkers` and `unsubscribe()` will cancel tasks of other `HandlerWorker`s too which is incorrect behavior.